src/main/php/CodacyPDepend/index.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace CodacyPDepend;
+
+require_once __DIR__ . '/../../../../vendor/autoload.php';
+require_once('CodacyConfiguration.php');
+require_once('CodacyReportGenerator.php');
+require_once('CodacyResult.php');
+
+use CodacyResult;
+use PDepend\Application;
+use Ds\Map;
+use LineComplexity;
+
+function genericMap($f, $list)
+{
+    $res = array();
+    foreach ($list as $e) {
+        array_push($res, $f($e));
+    }
+    return $res;
+}
+
+function pushToMapValueArray($map, $key, $value)
+{
+    if ($map->hasKey($key)) {
+        array_push($map[$key], $value);
+    } else {
+        $map[$key] = array($value);
+    }
+}
+
+function concatMapValueArray($map, $key, $array)
+{
+    if ($map->hasKey($key)) {
+        $map[$key] = array_merge($map[$key], $array);
+    } else {
+        $map[$key] = $array;
+    }
+}
+
+function mergeMapsOfArrays($res, $other)
+{
+    foreach ($other as $key => $value) {
+        concatMapValueArray($res, $key, $value);
+    }
+}
+
+function getFilename($content)
+{
+    $comp_unit = $content->getCompilationUnit();
+    return $comp_unit->getFileName();
+}
+
+function filenameToFunctions($node)
+{
+    $map = new Map();
+    foreach ($node->getFunctions() as $function) {
+        $key = getFilename($function);
+        pushToMapValueArray($map, $key, $function);
+    }
+
+    return $map;
+}
+
+function filenameToMethods($node)
+{
+    $map = new Map();
+    foreach ($node->getClasses() as $content) {
+        $key = getFilename($content);
+        concatMapValueArray($map, $key, $content->getMethods());
+    }
+    return $map;
+}
+
+function contentMapToFileComplexities($content_map, $cyclomaticAnalyzer)
```

Comment:
his files is just functions, we need unit tests.

Reply:
Can you elaborate here?

Reply:
yep, add unit tests for the functions in this file at least, please.

Reply:
Perfect, I'm doing it :) Fooled by the missing "T" in the message :-D